### ISSUE TITLE
How to link genomethreader and genemark to braker2

### ISSUE THREAD

**Issue by astulaaa** [bioconda/bioconda-recipes#18696]
Hello,

I installed BRAKER2 using conda, the packages genome threader and genemark could not be added to the same environment due to this error:  rm: cannot remove '/home/thiesgehrmann/.gm_key': No such file or directory
cp: cannot create regular file '/home/thiesgehrmann/.gm_key': No such file or directory

so I manually installed them in other directories, moved genemark key to recommended location. How exactly can I link these programs to BRAKER2? 

Thank you

**Comment by astulaaa** [comment]
I get identical issues "rm cannot remove gm_key" and "cp cannot create gm_key" when trying to add both genometreader and genemark to braker environment. Are these packages still functional despite these warnings/errors ?

**Comment by nextgenusfs** [comment]
I assume you tried the instructions that were printed to terminal after you install braker2 via conda?
```
The config/ directory from AUGUSTUS can be accessed with the variable AUGUSTUS_CONFIG_PATH.
BRAKER2 requires this directory to be in a writable location, so if that is not the case, copy this directory to a writable location, e.g.:
cp -r $AUGUSTUS_CONFIG_PATH /absolute_path_to_user_writable_directory/
export AUGUSTUS_CONFIG_PATH=/absolute_path_to_user_writable_directory/config

Due to license restrictions, both GeneMark and GenomeThreader should be additionally installed for BRAKER2 to fully work.
These packages can be either installed as part of the BRAKER2 environment, or the PATH variable should be configured to point to them.
The GeneMark key should be located in $HOME/.gm_key and GENEMARK_PATH should include the path to the GeneMark executables.
```
You should be able to install GeneMark via the directions from the developers and then set the environmental variable $GENEMARK_PATH to the location on your system telling the program where to look.  I've never installed genomethreader, so no help there.

**Comment by astulaaa** [comment]
Yes, I did follow the instructions provided by conda the locations are both writable and executable; genemark now can be detected, however, genomethreader still fails to be recognized. I tried numerous options :1) adding it to the path when installed outside conda:  export ALIGNMENT_TOOL_PATH=/Research/Programs/Genomethreader/gth-1.7.1-Linux_x86_64-64bit/bin/  2) adding it as braker option : 
--ALIGNMENT_TOOL_PATH=/Research/Programs/Genomethreader/gth-1.7.1-Linux_x86_64-64bit/bin/ 3) installing it using conda and exportinmg path to /bin from within conda environmen, 4) also as an option  --ALIGNMENT_TOOL_PATH=~/anaconda3/envs/bin/ and nothing seems to work. What could be the problem?

**Comment by NailouZhang** [comment]
just remove ENV/etc/conda/deactivate.d/genemark_es-confdir.sh and etc/conda/activate.d/genemark_es-confdir.sh can help you
